web/src/popover_menus.js

```diff
@@ -441,4 +466,66 @@ export function initialize() {
             message_actions_popover_keyboard_toggle = false;
         },
     });
+
+    // Starred messages popover
+    tippy_no_propagation(".starred-messages-sidebar-menu-icon", {
+        placement: "right",
+        maxWidth: "none",
+        popperOptions: {
+            modifiers: [
+                {
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: "bottom",
+                    },
+                },
+            ],
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            $popper.addClass("starred-messages-popover");
+
+            starred_messages_popover_instance = instance;
+
+            $popper.one("click", "#unstar_all_messages", (e) => {
+                e.preventDefault();
+                e.stopPropagation();
+                starred_messages_ui.confirm_unstar_all_messages();
+                instance.hide();
+                starred_messages_popover_displayed = false;
```

Comment:
Since we are calling `instance.hide()` we don't need to define `starred_messages_popover_displayed = false;` here since it is already handled in the `hidden` function defined for this.

Reply:
If the `instance.props.isShown` is present in the Tippy instance, it may not be necessary to use a local state at all, i assume.

Reply:
Yeah, but it is the same as tracking if `instance` of this popover is defined or not along with whether it is shown or not. Also, we don't track `instance` of other popovers right now.  I can imagine us simplifying this in the future, though.